primefaces/src/main/java/org/primefaces/renderkit/HeadRenderer.java

```diff
@@ -180,47 +182,11 @@ public void encodeEnd(FacesContext context, UIComponent component) throws IOExce
     }
 
     protected void encodeCSS(FacesContext context, String library, String resource) throws IOException {
-        ResourceHandler resourceHandler = context.getApplication().getResourceHandler();
-        if (resourceHandler.isResourceRendered(context, resource, library)) {
-            // resource already rendered, skip
-            return;
-        }
-
-        ResponseWriter writer = context.getResponseWriter();
-        ExternalContext externalContext = context.getExternalContext();
-
-        Resource cssResource = resourceHandler.createResource(resource, library);
-        if (cssResource == null) {
-            throw new FacesException("Error loading CSS, cannot find \"" + resource + "\" resource of \"" + library + "\" library");
-        }
-        else {
-            writer.startElement("link", null);
-            writer.writeAttribute("type", "text/css", null);
-            writer.writeAttribute("rel", "stylesheet", null);
-            writer.writeAttribute("href", externalContext.encodeResourceURL(cssResource.getRequestPath()), null);
-            writer.endElement("link");
-        }
+        ResourceUtils.addStyleSheetResource(context, library, resource);
```

Comment:
IMO adding real JSF resources during encoding sounds a bit late
why not system listener or phase listener?

Reply:
I tried both a SystemEventListener and a PhaseListener and the order was never correct.   Depending on where things were added locale could be added before `core.js` thus not loading the locales because `window.PrimeFaces` did not exist yet.  

Updating the HeadRenderer this way felt the "safest" and most backwards compatible to ensure if people are using the `<f:facet name="first" />` stuff it would all work as it does now it just no longer uses `<link>`

Reply:
That being said I was using `UiViewRoot` with `PostAddToView` so not sure if that was the right SystemComponentListener place to do it?